cms/djangoapps/contentstore/views/videos.py

```diff
@@ -236,6 +256,108 @@ def video_images_handler(request, course_key_string, edx_video_id=None):
     return JsonResponse({'image_url': image_url})
 
 
+def validate_transcript_preferences(
+        provider, cielo24_fidelity, cielo24_turnaround, three_play_turnaround, preferred_languages
+):
+    """
+    Validate 3rd Party Transcription Preferences.
+
+    Arguments:
+        provider: Transcription provider
+        cielo24_fidelity:  Cielo24 transcription fidelity.
+        cielo24_turnaround: Cielo24 transcription turnaround.
+        three_play_turnaround: 3PlayMedia transcription turnaround.
+        preferred_languages: list of language codes.
+
+    Returns:
+        validated preferences or a validation error.
+    """
+    error, preferences = None, {}
+
+    # validate transcription providers
+    transcription_plans = get_3rd_party_transcription_plans()
+    if provider in transcription_plans.keys():
+
+        # Further validations for providers
+        if provider == TranscriptProvider.CIELO24:
+
+            # Validate transcription fidelity
+            supported_fidelities = transcription_plans[provider]['fidelity']
+            if cielo24_fidelity in supported_fidelities.keys():
+
+                # Validate transcription turnaround
+                supported_turnarounds = transcription_plans[provider]['turnaround']
+                if cielo24_turnaround not in supported_turnarounds.keys():
+                    error = _('Invalid cielo24 turnaround.')
+                    return error, preferences
+
+                # Validate transcription languages
+                supported_languages = transcription_plans[provider]['fidelity'][cielo24_fidelity]['languages']
+                if not len(preferred_languages) or not (set(preferred_languages) <= set(supported_languages.keys())):
```

Comment:
What's the idea behind `(set(preferred_languages) <= set(supported_languages.keys())`  <= symbol i mean, if you're doing it as subset then it's wrong.